tests/test_objects.py

```diff
@@ -382,11 +386,44 @@ def test_external_api_aliases_failure(client, create_aliases_failure_patcher, da
     assert not resp.json().get("upload_url")
     assert not resp.json().get("aliases")
     assert not resp.json().get("metadata")
+    assert not resp.json().get("authz")
 
     assert create_aliases_failure_patcher["data_upload_mock"].called
     assert create_aliases_failure_patcher["create_aliases_mock"].called
 
 
+@respx.mock
+@pytest.mark.parametrize(
+    "data",
+    [
+        # 'upload' is not an allowed value for alias
+        {
+            "file_name": "test.txt",
+            "authz": {"version": 0, "resource_paths": ["/programs/DEV"]},
+            "aliases": ["upload", "alias2", "alias3"],
+            "metadata": {"foo": "bar"},
+        },
+    ],
+)
+def test_create_guid_unallowed_alias(client, valid_upload_file_patcher, data):
```

Comment:
I'm not sure if I've implemented the checks as intended. POST/objects/upload only has a check for alias. It does not check GUID. 
Should the POST/objects/{guid} or POST/metadata/{guid} have checks for GUID?

Reply:
Yes we should also make sure "alias" is not allowed as the GUID, because the `POST /api/v1/objects/{{GUID or ALIAS}}` endpoint accepts either an alias or a GUID and we want to avoid conflicts with `POST /api/v1/objects/upload` in both cases

Reply:
@george42-ctds i'm sorry, there was a typo in my previous message 🤦‍♀️ i meant `Yes we should also make sure "upload" is not allowed as the GUID`

Reply:
Oh, lol. The checks/tests have been updated for 'upload' instead of alias.